<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeolocalisationConfigWidgetData</class>
 <widget class="QWidget" name="GeolocalisationConfigWidgetData">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>471</width>
    <height>757</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Geolocation</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QLabel" name="label">
     <property name="styleSheet">
      <string notr="true">QLabel { font: bold; }</string>
     </property>
     <property name="text">
      <string>Select geolocation method</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QRadioButton" name="addressLocRB">
     <property name="text">
      <string>By &amp;address</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>&lt;html&gt;&lt;body&gt;&lt;p&gt;Use the &lt;a href=&quot;https://nominatim.openstreetmap.org/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0d47f1;&quot;&gt;Nominatim&lt;/span&gt;&lt;/a&gt; geocoding service to get the nodes locations corresponding to the values stored in the&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="openExternalLinks">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="addressLabel">
       <property name="styleSheet">
        <string notr="true">QLabel { font: bold; }</string>
       </property>
       <property name="text">
        <string>address property</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="tlp::TulipComboBox" name="addressPropCB">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>325</width>
         <height>16777215</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QCheckBox" name="automaticChoiceCB">
     <property name="toolTip">
      <string>&lt;p&gt;If checked, when multiple locations are retrieved the one whose the returned address gives the best match with the input address (i.e the one for whom the input address is found closest to its beginning) is choosen.&lt;br/&gt;If there is no possible match (the input address is not found in any of returned addresses) the first location is choosen.&lt;br/&gt;If not checked, a dialog is then displayed to allow the user to choose among the list of found locations.&lt;br/&gt;In any case, if only one location is retrieved, it is choosen.&lt;/p&gt;</string>
     </property>
     <property name="text">
      <string>Automatic choice</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="createLatLngPropsCB">
     <property name="toolTip">
      <string>If checked, a &quot;latitude&quot; and &quot;longitude&quot; properties are created to store the locations retrieved from the geocoding service</string>
     </property>
     <property name="text">
      <string>Store locations in latitude/longitude properties</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="resetLatLngValuesCB">
     <property name="toolTip">
      <string>If not checked, when latitude/longitude values are already set  they are not retrieved from geocoding service.</string>
     </property>
     <property name="text">
      <string>Recompute all latitude/longitude values</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QRadioButton" name="latLngRB">
     <property name="text">
      <string>By &amp;latitude/longitude</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Use the nodes locations stored in latitude/longitude properties</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="3">
      <widget class="tlp::TulipComboBox" name="latPropCB">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="latitudeLabel">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel { font: bold; }</string>
       </property>
       <property name="text">
        <string>latitude property</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="longitudeLabel">
       <property name="styleSheet">
        <string notr="true">QLabel { font: bold; }</string>
       </property>
       <property name="text">
        <string>longitude property</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="tlp::TulipComboBox" name="lngPropCB">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QFrame" name="edgesControlPointsGB">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="edgesPathsCB">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>edges pat&amp;hs</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="pathLabel">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>path:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="edgesPathsPropertyCB">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select a double vector property containing lists of &lt;br/&gt;latitude/longitude (in the form [lat1, lng1, lat2, lng2, ...]) &lt;br/&gt;defining paths for edges&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <spacer name="horizontalSpacer_6">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="genLayoutButton">
       <property name="text">
        <string>Generate Layout</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>tlp::TulipComboBox</class>
   <extends>QComboBox</extends>
   <header>tulip/TulipComboBox.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>latLngRB</sender>
   <signal>toggled(bool)</signal>
   <receiver>edgesControlPointsGB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel">
     <x>235</x>
     <y>384</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edgesPathsCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>edgesPathsPropertyCB</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>edgesPathsCB</sender>
   <signal>toggled(bool)</signal>
   <receiver>pathLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>36</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
